go/vt/vtgate/evalengine/expr_logical.go

```diff
@@ -100,55 +100,100 @@ func (left boolean) not() boolean {
 	}
 }
 
-func (left boolean) and(right boolean) boolean {
+func opAnd(le, re Expr, env *ExpressionEnv) (boolean, error) {
```

Comment:
These are all changed so we can also do lazy evaluation if the left hand side already gives a definitive answer. If the left hand side of an `AND` is already explicitly `false`, we never evaluable the right hand side (which matches MySQL behavior but was never tested before).